src/power_grid_model/data_types.py

```diff
@@ -13,60 +13,84 @@
 
 # When we're dropping python 3.8, we should introduce proper NumPy type hinting
 
-SparseBatchArray = Dict[str, np.ndarray]
+SingleArray = Union[np.ndarray]
```

Comment:
Is there a specific reason to use Union on a single type, Union[np.ndarray]? Does it make a difference?

Reply:
yes the problem is that np.ndarray not subclassable is. there's a way around that using `np.NDArray` but it does not completely solve our problem. Otherwise, we would need python3.10